crates/driver/src/infra/config/file/mod.rs

```diff
@@ -677,6 +673,42 @@ fn default_max_order_age() -> Option<Duration> {
     Some(Duration::from_secs(300))
 }
 
+#[serde_as]
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "kebab-case", deny_unknown_fields)]
+pub struct BadTokenDetectionConfig {
+    /// How long should the token quality computed by the simulation
+    /// based logic be cached.
+    #[serde(
+        with = "humantime_serde",
+        default = "default_simulation_bad_token_max_age"
+    )]
+    pub simulation_bad_token_max_age: Duration,
+
+    /// The ratio of failures to attempts that qualifies a token as unsupported.
+    #[serde(default = "default_metrics_bad_token_failure_ratio")]
+    pub metrics_bad_token_failure_ratio: f64,
+
+    /// The minimum number of attempts required before evaluating a token’s
+    /// quality.
+    #[serde(default = "default_metrics_bad_token_required_measurements")]
+    pub metrics_bad_token_required_measurements: u32,
```

Comment:
I think there should be 2 places to configure stuff.
The cache for the simulation based approach is shared across all solvers that want to use it so it needs to be a global setting for the whole driver.
The metrics based approach is specific to each solver. I imagine in practice we'll probably end up configuring the same parameters for all solvers that use this but conceptually it could be specific to each solver.